### PR TITLE
Hook up the rewind fixers on the AL UI

### PR DESCRIPTION
**Pull request by pidgeonman**
#### Changes proposed in this Pull Request

Hook up the rewind fixers to the UI, so that we can display the threats found by the Rewind security scanner in the Activity Log and let the user take actions, such as fix or ignore them. Right now the backend part is done and the frontend (Calypso) is partially completed.

#### Testing instructions

There are different use cases that will be supported as this PR progresses, so this description will be updated accordingly. **Only fixable alerts are supported at the moment.**

#### a) Vulnerable extensions alerts
1) Upload a known vulnerable plugin. I'm using the infamous facebook_for_woocommerce Version 1.9.12, get it from .org. There seems to be some (automated?) cleanup process in Atomic V2 that would eliminate any vulnerable plugin as soon as it's uploaded, so I've tested it on my own Pressable test site.
2) Trigger a backup. I've been using the cli jetpack_backups/bin/backup.js program, like: 

> node backup.js -b {your blog ID} --config local-transport

you should receive a security alert on your email, mobile app and Calypso ActivityLog. That is good. If you don't see any alert, it means that the scanner didn't pick it up, so try with a different plugin and repeat.
3) Make sure you're in DEV, so fire up your local Calypso, go to Activity, you should see the threat alert(s) at the top. Example URL: 

> http://calypso.localhost:3000/activity-log/{your_site_hostname}

4) Click on the pink **Fix** button at the right of the alert. A spinner will be displayed and the button will be disabled, indicating a process has been triggered.
You can also **Ignore** a threat by clicking that option in the dropdown box.
5) When the fix is completed, the list of alerts will be re-rendered and the fixed alert should be gone. It is rare, but this process could fail at the server side, which is outside the scope of this PR. Failure or error handling is not properly implemented yet, but you should see the error message in the console of your browser's Developer Tools.

### CODE REVIEW

File: client/my-sites/activity/activity-log/threat-alert.jsx [hunk 1]
```diff
@@ -150,12 +341,13 @@ export class ThreatAlert extends Component {
...
 										/>
 									</span>
+									{ this.getSpinner() }
 									<SplitButton
 										compact
 										primary
 										label={ translate( 'Fix threat' ) }
-										onClick={ () => debug( 'main button clicked' ) }
-										disabled={ false }
+										onClick={ this.fixThreat }
```
Comment (singerb) [inline client/my-sites/activity/activity-log/threat-alert.jsx]: Not all threats are fixable. Not sure what we want to do design-wise in that case though; can we disable the main button but still have the pop-over? Or should `Ignore` move into the main button area?

It'd also be nice to use the URL for fixing that's passed in via the API, which I think you're now getting with the `threat.actions` mapping?

File: client/my-sites/activity/activity-log/threat-alert.jsx [hunk 2]
```diff
@@ -28,6 +32,10 @@ import './threat-alert.scss';
 const debug = debugFactory( 'calypso:activity-log' );
 
 const detailType = threat => {
+	if ( threat.hasOwnProperty( 'table' ) ) {
```
Comment (singerb) [inline client/my-sites/activity/activity-log/threat-alert.jsx]: I think the table stuff should move out into it's own PR as part of that work? Probably just here still from the meetup work I'm guessing.

**Review by singerb — COMMENTED** [review]
Couple comments; can't speak to the more Calypso stuff, will leave that for Steve/Rich.

**Comment by seear** [comment]
I think if we move the fix & status requests up into the `rewind-alerts.jsx` component, we can have that component update on threat status change, which should cause [another request for alerts](https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/activity/activity-log/rewind-alerts.jsx#L41), removing any threats that are no longer active.

File: client/my-sites/activity/activity-log/threat-alert.jsx (hunk 1 above)
Comment (seear) [reply client/my-sites/activity/activity-log/threat-alert.jsx]: > use the URL for fixing that's passed in via the API

I think we found that that URL was wrong, possibly the namespace was wrong. Anyway it 404'd.

Comment (singerb) [reply client/my-sites/activity/activity-log/threat-alert.jsx]: Ah, so that should be fixed. It looks like `$this->rest_base` is in the wrong place in the URL; `site-alerts.php:get_site_alerts()`. The presence of that action is one indication that the threat can be fixed, although I believe we also have the `fixable` flag in the threat data too.

Comment (seear) [reply client/my-sites/activity/activity-log/threat-alert.jsx]: > that should be fixed

I'm not totally sold on the design of having the URL in the response, given that we can construct it (correctly) anyway, making it redundant.

Comment (aeryx) [reply client/my-sites/activity/activity-log/threat-alert.jsx]: >making it redundant.
I agree 1000%. Also, other requests everywhere on Calypso are built the same way as I've done it here, so using the URL from a service deviates the implementation from the main design.

Comment (aeryx) [reply client/my-sites/activity/activity-log/threat-alert.jsx]: >I think we found that that URL was wrong, possibly the namespace was wrong. Anyway it 404'd.

The URL is fine. It's just that the wpcom-http::http() method is quirky. I'll probably make a PR for that too.

Comment (aeryx) [reply client/my-sites/activity/activity-log/threat-alert.jsx]: >Not all threats are fixable. 

@singerb , could you please give me an example of a non-fixable threat?

File: client/my-sites/activity/activity-log/threat-alert.jsx (hunk 2 above)
Comment (aeryx) [reply client/my-sites/activity/activity-log/threat-alert.jsx]: Yes, this remained in my codebase from the meetup work. I'll take it out of this.

File: client/my-sites/activity/activity-log/threat-alert.jsx (hunk 1 above)
Comment (singerb) [reply client/my-sites/activity/activity-log/threat-alert.jsx]: As I said, I'm not wedded to doing it this way, especially if it's the only place in Calypso that does.

**Comment by singerb** [comment]
For a non-fixable threat, do something like add the EICAR test string to a PHP file in your uploads, or just `if ( false ) eval( $_POST['foo']; );`; re-backup the site and you should have a threat that has `fixable: false` in the threat data, as well as no URL for the fix action, just an ignore one (although it sounds like we may be deprecating those).

**Comment by seear** [comment]
This PR needs testing instructions in the description. Random example: https://github.com/Automattic/wp-calypso/pull/34568

File: client/state/data-getters/index.js [hunk 3]
```diff
@@ -101,8 +101,9 @@ export const requestGeoLocation = () =>
 		{ fromApi: () => ( { body: { country_short } } ) => [ [ 'geo', country_short ] ] }
 	);
 
-export const requestSiteAlerts = siteId => {
+export const requestSiteAlerts = ( siteId, force ) => {
```
Comment (seear) [inline client/state/data-getters/index.js]: Some other functions in this module have freshness as a parameter, for example:

`export const requestActivityLogs = ( siteId, filter, { freshness = 5 * 60 * 1000 } = {} ) => {`

It doesn't look like they are ever used, but we may as well stick to the same pattern, passing in 0 when we want to force an update.

File: client/my-sites/activity/activity-log/rewind-alerts.jsx [hunk 4]
```diff
@@ -18,27 +18,42 @@ import { requestSiteAlerts } from 'state/data-getters';
  */
 import './rewind-alerts.scss';
 
-function RewindAlerts( { alerts } ) {
-	const translate = useTranslate();
+let refreshFlag = false;
```
Comment (seear) [inline client/my-sites/activity/activity-log/rewind-alerts.jsx]: I think `refreshFlag` can be component state instead of a local variable, then the site alerts will be fetched whenever it changes.

If we pass a function to set `refreshFlag` into `<ThreatAlert>`, we can call it whenever `checkFixStatus` returns, which will prompt this component to re-fetch all the alerts.

I'm getting very uneasy with this, however, because this kind of action->data->update flow should go through redux state.

**Comment by rcoll** [comment]
I gave this a good test, and while it does make a network call, it doesn't seem to do anything. Or perhaps it's showing the `fix` button for an unfixable threat. Not sure. Let me know when you're online @pidgeonman and we can sync up on this.

**Comment by seear** [comment]
> I'm getting very uneasy with this, however, because this kind of action->data->update flow should go through redux state.

Looking at this PR, and thinking about it further, I think we need to make this change properly and store the alerts in Redux state, so that when we fix an alert the UI updates without us needing to make any hacks to bypass the Redux data flow.

@pidgeonman and @rcoll, can you work together to store the the alert state in Redux? It probably will make sense to do that in a separate PR.
